packages/emotion-theming/types/index.d.ts

```diff
@@ -3,30 +3,43 @@
 
 import * as React from 'react'
 
-import { AddOptionalTo, PropsOf } from './helper'
+import {
+  StyledComponent,
+  StyledOptions,
+  CreateStyledComponent,
+  StyledTags
+} from '@emotion/styled'
+import { PropsOf, Omit } from '@emotion/styled-base'
 
 export interface ThemeProviderProps<Theme> {
   theme: Partial<Theme> | ((outerTheme: Theme) => Theme)
   children?: React.ReactNode
 }
 
-export function ThemeProvider<Theme>(
-  props: ThemeProviderProps<Theme>
-): React.ReactElement
-
-export function useTheme<Theme>(): Theme
+export interface ThemeProvider<Theme = any> {
```

Comment:
shouldn't this be both restricted to `{}` and defaulted to it? `any` doesn't seem right - especially that there are places where we spread themes:
https://github.com/emotion-js/emotion/blob/1b3791d4c458838a097c815a38108c9719023b92/packages/emotion-theming/src/theme-provider.js#L30

Reply:
Interesting. I can see how we could restrict on ThemeProvider, but for most other scenarios I think it needs to be `any` unless emotion definitely ensures it will pass an empty object for theme and never `undefined`

Reply:
Yes - this is given. The empty object is a default:
 https://github.com/emotion-js/emotion/blob/eabb4609729464fb37d58fa02d370caa7eca511f/packages/core/src/context.js#L17

Reply:
Finally sorted!